src/services/chat/api.ts

```diff
@@ -109,6 +125,28 @@ const pushMessages = async (
   });
 };
 
+const setIsTyping = async (
+  chatRef: DocumentRefData,
+  userRef: DocumentRefData,
+  isTyping: boolean,
+): Promise<void> => {
+  await firestore().runTransaction(async (t) => {
+    const chat = await t.get(chatRef);
+    const currentlyTypingArray = chat.data()?.currentlyTyping as DocumentRefData[];
+    if (currentlyTypingArray) {
```

Comment:
Sugestion: Instead of checking if `currentlyTypingArray` is defined, you could change the create chat function to start this field as an empty array. That way you could remove the outer if-else-if from the code (theoretically).

Reply:
Changed